agent/engine/docker_task_engine_test.go

```diff
@@ -156,6 +156,7 @@ func TestBatchContainerHappyPath(t *testing.T) {
 	default:
 	}
 	eventsReported.Wait()
+	client.EXPECT().DescribeContainer(gomock.Any()).AnyTimes()
```

Comment:
I think there are bigger issues in this test than the indeterminate nature of steady state check being invoked. IIUC, Line 135 overrides Line 134, so I'm not even sure that the steady state check is happening here. I think that we should modify the test to wait/block for that to happen.